### PR TITLE
Made thread pool optional and configurable

### PR DESCRIPTION
**Pull request by whiskeysierra**
### riptide-core

Executor is now optional and users can configure a `ClientHttpRequestFactory` directly. Not configuring an executor yields synchronous operations, i.e. futures will already be completed.

### riptide-spring-boot-autoconfigure

`threads` got an `enabled` flag (default: `true`) which controls whether a thread pool is configured or not.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: riptide-core/src/main/java/org/zalando/riptide/DefaultHttpBuilder.java [hunk 1]
```diff
@@ -129,22 +127,23 @@ public ConfigurationStage defaultPlugins() {
...
 
     @Override
     public Http build() {
         final List<HttpMessageConverter<?>> converters = converters();
-
         final List<Plugin> plugins = new ArrayList<>();
-        plugins.add(new AsyncPlugin(executor));
+
+        if (executor != null) {
+            plugins.add(new AsyncPlugin(executor));
```
Comment (whiskeysierra) [inline riptide-core/src/main/java/org/zalando/riptide/DefaultHttpBuilder.java]: Next step could be to remove the ability to configure an executor from the API and expose this plugin directly (e.g. a `ThreadPoolPlugin`) which could then actually be moved outside of the core (`riptide-concurrent`?), together with the content of the `org.zalando.riptide.concurrent` package.

Comment (whiskeysierra) [reply riptide-core/src/main/java/org/zalando/riptide/DefaultHttpBuilder.java]: Leaves the question of how to call the plugin phase that is currently called `aroundAsync`.

File: README.md [hunk 2]
```diff
@@ -244,12 +244,12 @@ The following implementations offer non-blocking IO:
 - [`Netty4ClientHttpRequestFactory`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/client/Netty4ClientHttpRequestFactory.html), using [Netty](https://netty.io/)
 - [`HttpComponentsAsyncClientHttpRequestFactory`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/client/HttpComponentsAsyncClientHttpRequestFactory.html), using [Apache HTTP Async Client](https://hc.apache.org/httpcomponents-asyncclient-4.1.x/index.html)
 
-Non-blocking IO is asynchronous by nature. In order to provide asynchrony for blocking IO you need to register an executor. For synchronous operations it's possible to just pass `Runnable::run`.
+Non-blocking IO is asynchronous by nature. In order to provide asynchrony for blocking IO you need to register an executor. Not passing an executor will make all network communication synchrnonous, i.e. all futures returned by Riptide will already be completed.
```
Comment (lukasniemeier-zalando) [inline README.md]:
```suggestion
Non-blocking IO is asynchronous by nature. In order to provide asynchrony for blocking IO you need to register an executor. Not passing an executor will make all network communication synchronous, i.e. all futures returned by Riptide will already be completed.
```

File: riptide-core/src/main/java/org/zalando/riptide/DefaultHttpBuilder.java (hunk 1 above)
Comment (lukasniemeier-zalando) [reply riptide-core/src/main/java/org/zalando/riptide/DefaultHttpBuilder.java]: `aroundRiptide` :wink:

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**